core/test/unit/middleware/cors_spec.js

```diff
@@ -137,4 +141,41 @@ describe('cors', function () {
 
         done();
     });
+
+    it('should be enabled if the origin matches config.url', function (done) {
+        var origin = 'http://my.blog';
+        configUtils.set({
+            url: origin
+        });
+
+        req.get = sinon.stub().withArgs('origin').returns(origin);
+        res.get = sinon.stub().withArgs('origin').returns(origin);
+        req.headers.origin = origin;
+
+        cors(req, res, next);
+
+        next.called.should.be.true();
+        res.headers['Access-Control-Allow-Origin'].should.equal(origin);
+
+        done();
+    });
```

Comment:
> I'm not sure if that's what the test reflects?

I think it does? The ghost instance URL is being configured to the same thing as the request's `Origin` header. That's same-origin and we shouldn't reflect the header back in the response in that case.

> It seems to me that nothing in the test or the middleware (not even the cors middleware itself, which is surprising) does any checks to determine if the request is cross-origin

Nope. It's all based on the `Origin` header and matching it against a URL approved for cross-domain requests.

Really the only things the server needs to concern itself with are:
* Did I receive a request with an `Origin` header? If so, does it match my whitelist of approved origins? If yes, then reflect the origin in the `Access-Control-Allow-Origin` header.

In the event that an `Origin` comes in and it happens to be a same-origin request, the origin won't be in the whitelist and the `Access-Control-Allow-Origin` header won't be present in the response. However, this is okay because the browser that sent the request doesn't need it because the request is same-origin.

Sorry if this is confusing 😦 

> As far as I understood, whenever I made a same-origin request, origin wasn't present in the headers

Most browsers send `Origin` on same-origin requests but there are exceptions.